inc/c_util/tarray_ll.h

```diff
@@ -0,0 +1,183 @@
+    if (tarray == NULL)                                                                             \
+    {                                                                                               \
+        LogError("invalid arguments " MU_TOSTRING(TARRAY_TYPEDEF_NAME(T)) " * tarray=%p",           \
+            tarray);                                                                                \
+    }                                                                                               \
+    else                                                                                            \
+    {                                                                                               \
+        free((void*)tarray->arr);                                                                   \
+    }                                                                                               \
+}                                                                                                   \
+
+/*introduces a function definition for tarray_create*/
+#define TARRAY_LL_CREATE_DEFINE(C, T)                                                                                       \
+TARRAY(T) TARRAY_LL_CREATE(C)(void)                                                                                         \
+{                                                                                                                           \
+    TARRAY_TYPEDEF_NAME(T)* result;                                                                                         \
+    /*Codes_SRS_TARRAY_02_001: [ TARRAY_CREATE(T) shall call THANDLE_MALLOC to allocate the result. ]*/                     \
+    result = THANDLE_MALLOC(TARRAY_TYPEDEF_NAME(C))(TARRAY_LL_FREE_NAME(C));                                                \
+    if(result == NULL)                                                                                                      \
+    {                                                                                                                       \
+        LogError("failure in malloc(" MU_TOSTRING(TARRAY_TYPEDEF_NAME(T)) "=%zu", sizeof(TARRAY_TYPEDEF_NAME(T)));          \
+        /*return as is*/                                                                                                    \
+    }                                                                                                                       \
+    else                                                                                                                    \
+    {                                                                                                                       \
+        /*Codes_SRS_TARRAY_02_002: [ TARRAY_CREATE(T) shall call malloc to allocate result->arr. ]*/                        \
+        result->arr = malloc(1 * sizeof(T));                                                                                \
```

Comment:
> malloc(1 * sizeof(T));   [](http://example.com/codeflow?start=21&length=25)

unrelated to splitting the LL I guess, but why not have the  caller give an initial capacity to avoid allocating an array of 1 first if the initial size is known?

Reply:
It is a good suggestion, I will take it with another task. Here's the task: 

Task 11115085: TARRAY_CREATE can take an initial capacity

([https://msazure.visualstudio.com/One/_workitems/edit/11115085](https://msazure.visualstudio.com/One/_workitems/edit/11115085))